:memo: readme: Replace placeholders and add variable descriptions

LeSpocky opened the pull request:
First version of documentation.  Based on the template generated by `ansible-galaxy` back then.  (Almost) all variables described, at least shortly.  Fixes #10.

LeSpocky commented:
Lacks a useful description for variable `docker_v6_cidr`.  Open for feedback.

teuserer reviewed (approved):
Looks good to me

penguineer commented on README.md:
```diff
@@ -23,38 +23,110 @@ Ansible Role for installing the Docker runtime environment on your (Debian) host
 ## Requirements
 
-*nothing special*
+This is intended for Debian GNU/Linux based systems.
```
Please don't use a plain "this" at the beginning of a paragraph, as unnecessary ambiguity is introduced.

penguineer commented on README.md:
```diff
@@ -23,38 +23,110 @@ Ansible Role for installing the Docker runtime environment on your (Debian) host
 
-*nothing special*
+This is intended for Debian GNU/Linux based systems.
+Plain old Debian works, Ubuntu should work, others might work.
```
"The role has been tested on Debian and is expected to work on Ubuntu. Other distribution have not been tested."

penguineer commented on README.md:
```diff
@@ -23,38 +23,110 @@ Ansible Role for installing the Docker runtime environment on your (Debian) host
 
-This role can be installed through your *requirements.yml*, either from
-[Ansible Galaxy](https://galaxy.ansible.com/) or through the Git repo.
+This role can be installed through your *requirements.yml*,
```
Please only document the current status of the software. Future plans are part of an optional "roadmap" paragraph.

penguineer commented on README.md:
```diff
@@ -23,38 +23,110 @@ Ansible Role for installing the Docker runtime environment on your (Debian) host
-Any variables that are read from other roles and/or the global scope
-(ie. hostvars, group vars, etc.) should be mentioned here as well.*
+This role has currently no mandatory variables.
+You can just go with the defaults.
```
"You can go with just the defaults."

penguineer commented on README.md:
```diff
@@ -23,38 +23,110 @@ Ansible Role for installing the Docker runtime environment on your (Debian) host
-*A list of other roles hosted on Galaxy should go here, plus any details
-in regards to parameters that may need to be set for other roles, or
-variables that are used from other roles.*
+*Ansible builtin modules only.*
```
"No external dependencies", and maybe the Ansible version this role has been tested on, i.e. "Tested on Ansible x.z.z"

penguineer commented on README.md:
```diff
@@ -23,38 +23,110 @@ Ansible Role for installing the Docker runtime environment on your (Debian) host
 
-*Including an example of how to use your role (for instance, with
-variables passed in as parameters) is always nice for users too:*
+Just include the role and set some [variables](#role-variables).
```
Please leave this sentence out.

LeSpocky commented:
Addressed all review comments and force-pushed the changed version.  Removed WIP status.  Still lacks docs for `docker_v6_cidr`.  I don't know what it does.  Please assist.

penguineer commented on README.md:
```diff
@@ -23,38 +23,106 @@ Ansible Role for installing the Docker runtime environment on your (Debian) host
+      here to notice signing key changes.
+* `docker_apt_key_url`:
+    * Default: https://download.docker.com/linux/debian/gpg
+    * Description: Address to download the OpenPGP key, which is used to
```
"URL of the OpenGPG key"

penguineer commented on README.md:
```diff
@@ -23,38 +23,106 @@ Ansible Role for installing the Docker runtime environment on your (Debian) host
+    * Default: https://download.docker.com/linux/debian
+    * Description: Address of the third party apt repo to download
+      Debian packages from.
+      You might want to set this to the address of a local apt proxy
```
Here, too, "URL" instead of "address"

(side-nit: The term URL is outdated. In research only URI is used, as most URLs are opaque, i.e. mask the real location, and are technically URIs. However, the term is still used in administration, so it might be confusing to write URI here.)

penguineer commented on README.md:
```diff
@@ -23,38 +23,106 @@ Ansible Role for installing the Docker runtime environment on your (Debian) host
+      like *approx*, *apt-cacher* or the like.
+* `docker_cron_image_prune`:
+    * Default: `false`
+    * Description: Cleanup old images, which are not used anymore.
```
No comma here, better "Clean-up images without a container reference"

penguineer reviewed (approved):
I missed some things in the first go. They are not mandatory, but would improve quality. Therefore I accept the PR, but add some comments for possible changes.

LeSpocky commented:
> > Addressed all review comments and force-pushed the changed version. Removed WIP status. Still lacks docs for `docker_v6_cidr`. I don't know what it does. Please assist.
> 
> Used [here](https://github.com/netz39/ansible-role-host-docker/blob/2ac3597ab320532742061ad65b10e753551535f3/templates/daemon.json.j2#L6) to enable IPv6 for Docker. See https://docs.docker.com/engine/daemon/ipv6/#use-ipv6-for-the-default-bridge-network

I saw the place where it is used. Don't understand however what "enable IPv6 for Docker" means. It's not clear from their docs as well. Could mean anything. If you have a good suggestion, please add a second commit on this branch for the same PR.
